### PR TITLE
fix(BACK-1691): consider base token oracle state feasibility for WooFiV2

### PR DESCRIPTION
**Pull request by 0xNazarii**

### CODE REVIEW

File: src/dex/woo-fi-v2/woo-fi-v2-math.ts [hunk 1]
```diff
@@ -97,6 +113,14 @@ export class WooFiV2Math {
       let state1: TokenState = this.state.tokenStates[baseToken1];
       let state2: TokenState = this.state.tokenStates[baseToken2];
 
+      if (!state1.woFeasible || !state2.woFeasible) {
+        const token = !state1.woFeasible ? baseToken1 : baseToken2;
+        this.logger.error(
+          `Oracle state for base token ${token} is not feasible`,
+        );
+        return null;
```
Comment (KanievskyiDanylo) [inline src/dex/woo-fi-v2/woo-fi-v2-math.ts:121]: Both tokens can have `woFeasible` false, would be nice to log all of them

**Review by aburkut — APPROVED** [review]

File: src/dex/woo-fi-v2/woo-fi-v2-math.ts [hunk 2]
```diff
@@ -46,6 +46,14 @@ export class WooFiV2Math {
   _querySellBase(baseToken: Address, baseAmounts: bigint[]): bigint[] | null {
     try {
       const _state = this.state.tokenStates[baseToken];
+
+      if (!_state.woFeasible) {
+        this.logger.error(
```
Comment (KanievskyiDanylo) [inline src/dex/woo-fi-v2/woo-fi-v2-math.ts:51]: Can we also log `this.state.oracleTimestamp` to ease debugging for such cases?

**Review by KanievskyiDanylo — APPROVED** [review]

File: src/dex/woo-fi-v2/woo-fi-v2-math.ts (hunk 2 above)
Comment (0xNazarii) [reply src/dex/woo-fi-v2/woo-fi-v2-math.ts:51]: done

File: src/dex/woo-fi-v2/woo-fi-v2-math.ts (hunk 1 above)
Comment (0xNazarii) [reply src/dex/woo-fi-v2/woo-fi-v2-math.ts]: done

**Pull request merged**